File: app/src/main/java/net/imatruck/dsmanager/fragments/EditTaskDialog.java
package net.imatruck.dsmanager.fragments;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatDialogFragment;
import android.support.v7.widget.LinearLayoutCompat;
import android.widget.EditText;

import net.imatruck.dsmanager.R;


public class EditTaskDialog extends AppCompatDialogFragment {

    public interface EditTaskListener {
        void onDialogConfirmMove(String newDestination);
        void onDialogCancelMove();
    }

    EditTaskListener mListener;

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        float defaultMargin = getActivity().getResources().getDimension(R.dimen.dialog_margin);

        final LinearLayoutCompat container = new LinearLayoutCompat(getActivity());
        LinearLayoutCompat.LayoutParams containerParams = new LinearLayoutCompat.LayoutParams(
                LinearLayoutCompat.LayoutParams.MATCH_PARENT,
                LinearLayoutCompat.LayoutParams.WRAP_CONTENT);
        container.setLayoutParams(containerParams);

        final EditText newDestinationEditText = new EditText(getActivity());
        LinearLayoutCompat.LayoutParams lp = new LinearLayoutCompat.LayoutParams(
                0, LinearLayoutCompat.LayoutParams.WRAP_CONTENT, 1.0f);

        lp.setMargins((int) defaultMargin, 0, (int) defaultMargin, 0);

        newDestinationEditText.setLayoutParams(lp);

        container.addView(newDestinationEditText);

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        builder.setTitle(R.string.edit_task_dialog_title)
                .setMessage(R.string.edit_task_dialog_message)
                .setView(container)
                .setPositiveButton(R.string.edit_task_dialog_confirm,
                        new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        mListener.onDialogConfirmMove(newDestinationEditText.getText().toString());
                    }
                })
                .setNegativeButton(R.string.edit_task_dialog_cancel,
                        new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        mListener.onDialogCancelMove();
                    }
                })
                .setOnDismissListener(new DialogInterface.OnDismissListener() {
                    @Override
                    public void onDismiss(DialogInterface dialog) {
                        mListener.onDialogCancelMove();
                    }
                });

        return builder.create();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try {
            mListener = (EditTaskListener) context;
        } catch (ClassCastException cce) {
            throw new ClassCastException(context.toString() + " must implement EditTaskListener");
        }
    }
}

File: app/src/main/java/net/imatruck/dsmanager/fragments/EditTaskDialog.kt
package net.imatruck.dsmanager.fragments

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatDialogFragment
import android.support.v7.widget.LinearLayoutCompat
import android.widget.EditText

import net.imatruck.dsmanager.R


class EditTaskDialog : AppCompatDialogFragment() {

    interface EditTaskListener {
        fun onDialogConfirmMove(newDestination: String)
        fun onDialogCancelMove()
    }

    internal var mListener: EditTaskListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val defaultMargin = activity.resources.getDimension(R.dimen.dialog_margin)

        val container = LinearLayoutCompat(activity)
        val containerParams = LinearLayoutCompat.LayoutParams(
                LinearLayoutCompat.LayoutParams.MATCH_PARENT,
                LinearLayoutCompat.LayoutParams.WRAP_CONTENT)
        container.layoutParams = containerParams

        val newDestinationEditText = EditText(activity)
        val lp = LinearLayoutCompat.LayoutParams(
                0, LinearLayoutCompat.LayoutParams.WRAP_CONTENT, 1.0f)

        lp.setMargins(defaultMargin.toInt(), 0, defaultMargin.toInt(), 0)

        newDestinationEditText.layoutParams = lp

        container.addView(newDestinationEditText)

        val builder = AlertDialog.Builder(activity)

        builder.setTitle(R.string.edit_task_dialog_title)
                .setMessage(R.string.edit_task_dialog_message)
                .setView(container)
                .setPositiveButton(R.string.edit_task_dialog_confirm
                ) { _, _ -> mListener!!.onDialogConfirmMove(newDestinationEditText.text.toString()) }
                .setNegativeButton(R.string.edit_task_dialog_cancel
                ) { _, _ -> mListener!!.onDialogCancelMove() }
                .setOnDismissListener { mListener!!.onDialogCancelMove() }

        return builder.create()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        try {
            mListener = context as EditTaskListener
        } catch (cce: ClassCastException) {
            throw ClassCastException(context!!.toString() + " must implement EditTaskListener")
        }

    }
}
